src/class/object_custom_viewlists.php

```diff
@@ -766,6 +775,28 @@ private function getCustomColsForCategory ($sCategory)
 
 
 
+    private function getCustomColQuery ($sCategory, $sSelectQuery='')
+    {
+        // Return string with custom column names for object type $sCategory
+        // that can be used in the SELECT listing of an SQL query.
+        // $sSelectQuery can be given as an existing SELECT clause to avoid
+        // duplication of column names are already listed.
+        $aCustomColNames = array_keys($this->getCustomColsForCategory($sCategory));
+        if (count($aCustomColNames) > 0) {
+            $aNamesAggregated = array_map(function ($sName) use ($sSelectQuery) {
+                if (strpos($sName, $sSelectQuery) === false) {
+                    return 'MIN(`' . $sName . '`) AS `' . $sName . '`';
+                }
+                return '';
+            }, $aCustomColNames);
+            return join(', ', $aNamesAggregated);
+        }
+        return '';
+    }
```

Comment:
Sorry, it's more complicated than necessary, and now that I see it again I spotted a bug :sweat_smile: 

I've committed a fix in ee972e4 which is easier to read.